EventBus.Kafka/KafkaEventBusFactory.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using Finaps.EventBus.Kafka.Configuration;
+using Finaps.EventBus.Core;
+using Finaps.EventBus.Core.Abstractions;
+using Finaps.EventBus.Core.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using EventBus.Kafka;
+
+namespace Finaps.EventBus.Kafka
+{
+  public static class KafkaFactory
+  {
+    public static IEventBus Create(IServiceProvider serviceProvider, List<EventSubscription> subscriptions, KafkaOptions? options = null, ILoggerFactory? loggerFactory = null)
+    {
+      loggerFactory ??= new NullLoggerFactory();
+      options ??= new KafkaOptions();
+
+      var subscriptionsManager = new InMemoryEventBusSubscriptionsManager();
+      var publisher = CreatePublisher(options, loggerFactory);
+      var consumer = CreateSubscriber(options, loggerFactory);
+      var logger = loggerFactory.CreateLogger<IEventBus>();
+      var eventBus = new KafkaEventBus(publisher, consumer, subscriptionsManager, serviceProvider, logger);
+      foreach (var subscription in subscriptions)
+      {
+        eventBus.AddSubscription(subscription);
+      }
```

Comment:
Likely no longer needed for this implementation?

Reply:
Correct

Reply:
Nevermind, its needed for the eventbus handlers